lib/Optimizer/Transforms/QuakeSynthesizer.cpp

```diff
@@ -617,6 +623,19 @@ class QuakeSynthesizer
         continue;
       }
 
+      if (auto charSpanTy = dyn_cast<cudaq::cc::CharspanType>(type)) {
+        const char *ptrToSizeInBuffer =
+            static_cast<const char *>(args) + offset;
+        auto sizeFromBuffer =
+            *reinterpret_cast<const std::uint64_t *>(ptrToSizeInBuffer);
+        std::size_t bytesInType = sizeof(char);
+        auto vectorSize = sizeFromBuffer / bytesInType;
+        stdVecInfo.emplace_back(
```

Comment:
Building these strings on the stack in assembly code is very inefficient, and we would never expect these to be inlined like a vector of rotation angles, for example.

Let me see if I can't determine why this was changes to a vector. That seems like a mistake that's rippling through the compiler and causing many problems.

Reply:
> Building these strings on the stack in assembly code is very inefficient, and we would never expect these to be inlined like a vector of rotation angles, for example.
> 
> Let me see if I can't determine why this was changed to a vector. That seems like a mistake that's rippling through the compiler and causing many problems.

Offline conversations led to the conclusion that we should merge this as is for now and just know that we are eventually going to go back to a Pauli word being a string rather than a vector. @schweitzpgi - please confirm by marking this conversation as resolved, or please add additional comments as you see fit.